tensorflow/lite/tools/optimize/quantize_model.h

```diff
@@ -65,6 +65,26 @@ TfLiteStatus QuantizeModel(flatbuffers::FlatBufferBuilder* builder,
                            const std::unordered_set<string>& operator_names,
                            ErrorReporter* error_reporter);
 
+// Same as above, but enables to provide activation type, which
+// could be TensorType_INT16 or TensorType_INT8.
+//
+// Note: This is a private API, subject to change.
+TfLiteStatus QuantizeModel(flatbuffers::FlatBufferBuilder* builder,
+                           ModelT* model, const TensorType& input_type,
+                           const TensorType& output_type, bool allow_float,
+                           const TensorType& activations_type,
+                           ErrorReporter* error_reporter);
+
+// Quantizes input_model and populates the provided builder with the new model
+// with all possible input parameters.
+// All functions above call this function underneath.
+TfLiteStatus QuantizeModel(flatbuffers::FlatBufferBuilder* builder,
+                           ModelT* model, const TensorType& input_type,
```

Comment:
I had to move out this function from quantize_model.cc, because I had troubles with tests:
for example, when {} is passed as const std::unordered_set<string>& operator_names, 
C++ uses the new version 
 TfLiteStatus QuantizeModel(flatbuffers::FlatBufferBuilder* builder,
                           ModelT* model, const TensorType& input_type,
                           const TensorType& output_type, bool allow_float,
                           const TensorType& activations_type,
                           ErrorReporter* error_reporter);
This is because it converts {} to TensorType.
It could be fixed by replacing enum TensorType to enum class TensorType, but I think this is a better solution.

Reply:
instead, could you update the existing function signatures, since these functions technically do not have backwards compat guarantees.